plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/dialog/SelectRouteDialog.java

```diff
@@ -28,7 +32,30 @@
 	public SelectRouteDialog(List<IRoute> routes, Shell shell) {
 		super(shell, new RouteLabelProvider());
 		setTitle("Select Route");
-		setMessage("Select the route to open in a browser.");
+
+		StringBuilder message = new StringBuilder();
+		if(!routes.isEmpty()) {
+			IRoute route = routes.get(0);
+			IConnection connection = ConnectionsRegistryUtil.safeGetConnectionFor(route);
+			if(connection != null) {
+				message.append("Server: ")
+				.append(connection.getUsername()).append(" ").append(connection.getHost())
+				.append(StringUtils.getLineSeparator());
+			}
+			IProject project = route.getProject();
```

Comment:
Since all you need is the project name, you could use #getNamespace() from the route.  It will return the same value.  Additionally it will not make a call to grab the project since we dont cache this info at the moment